### ISSUE TITLE
CMakeLists vs Makefile

### ISSUE THREAD

**Issue by grische** [ComputationalRadiationPhysics/cuda_memtest#22]
Why do you have both a `CMakeLists.txt` and a `Makefile`? Which one is the preferred one?

Considering that the `Makefile` is being overriden by cmake, can it be removed to avoid ambiguity?

**Comment by ax3l** [comment]
Hi and thanks for the report!

We maintain the `CMakeLists.txt` version. The old `Makefile` script is just lingering around from the long-term unmaintained repo that we forked from: https://sourceforge.net/p/cudagpumemtest

We should remove it accordingly, but it probably contains some hints in case someone actually wants to use the OpenCL version (instead of the CUDA tests that we focus on).

**Comment by ax3l** [comment]
I personally also added a [Spack package](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/cuda-memtest/package.py) that builds from CMake, as recommended.

**Comment by ax3l** [comment]
> Considering that the Makefile is being overriden by cmake, can it be removed to avoid ambiguity?

Don't build CMake projects in-source, please. Create a temporary build directory and execute `cmake <pathToSource>` in it.

**Comment by grische** [comment]
@ax3l Makes sense. 

Could you add build instructions to the README? I assumed the cmake variant is the new one, but the readme clearly stated that the Makefile is the current version which was quite confusing.

**Comment by ax3l** [comment]
Yes, good idea. We initially thought our changes might be accepted upstream again and tried to change as little as possible.

I think a decade later we can assume nothing will happen upstream anymore and so let's change more boldly along the repo lines ;)
